#!/usr/bin/env python3
PACKAGE = "opencv_final"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# 检测颜色选择
gen.add("enemy_color", int_t, 0, "Target color (0:Red, 1:Blue, 2:Orange)", 2, 0, 2)

# 红色阈值参数
gen.add("red_hue_min", int_t, 0, "Red Hue Min", 0, 0, 180)
gen.add("red_hue_max", int_t, 0, "Red Hue Max", 10, 0, 180)
gen.add("red_sat_min", int_t, 0, "Red Saturation Min", 100, 0, 255)
gen.add("red_sat_max", int_t, 0, "Red Saturation Max", 255, 0, 255)
gen.add("red_val_min", int_t, 0, "Red Value Min", 50, 0, 255)
gen.add("red_val_max", int_t, 0, "Red Value Max", 255, 0, 255)

# 蓝色阈值参数
gen.add("blue_hue_min", int_t, 0, "Blue Hue Min", 100, 0, 180)
gen.add("blue_hue_max", int_t, 0, "Blue Hue Max", 130, 0, 180)
gen.add("blue_sat_min", int_t, 0, "Blue Saturation Min", 100, 0, 255)
gen.add("blue_sat_max", int_t, 0, "Blue Saturation Max", 255, 0, 255)
gen.add("blue_val_min", int_t, 0, "Blue Value Min", 50, 0, 255)
gen.add("blue_val_max", int_t, 0, "Blue Value Max", 255, 0, 255)

# 橙色阈值参数
gen.add("orange_hue_min", int_t, 0, "Orange Hue Min", 10, 0, 180)
gen.add("orange_hue_max", int_t, 0, "Orange Hue Max", 25, 0, 180)
gen.add("orange_sat_min", int_t, 0, "Orange Saturation Min", 150, 0, 255)
gen.add("orange_sat_max", int_t, 0, "Orange Saturation Max", 255, 0, 255)
gen.add("orange_val_min", int_t, 0, "Orange Value Min", 180, 0, 255)
gen.add("orange_val_max", int_t, 0, "Orange Value Max", 255, 0, 255)

# 装甲板参数
gen.add("min_area", double_t, 0, "Minimum armor area", 500, 0, 5000)
gen.add("max_area", double_t, 0, "Maximum armor area", 2500, 0, 5000)
gen.add("min_aspect_ratio", double_t, 0, "Minimum aspect ratio", 0.3, 0.1, 5.0)
gen.add("max_aspect_ratio", double_t, 0, "Maximum aspect ratio", 1.1, 0.1, 5.0)
gen.add("min_contour_area", double_t, 0, "Minimum contour area", 100, 0, 1000)

# 形态学参数
gen.add("morph_open_size", int_t, 0, "Morphology open kernel size", 3, 0, 10)
gen.add("morph_close_size", int_t, 0, "Morphology close kernel size", 5, 0, 10)

exit(gen.generate(PACKAGE, "opencv_final", "ArmorDetector"))